inbound_test.go

```diff
@@ -142,3 +142,27 @@ func TestInboundConnection_CallOptions(t *testing.T) {
 		require.NoError(t, err, "Call through proxy failed")
 	})
 }
+
+func TestBlackhole(t *testing.T) {
+	testutils.WithTestServer(t, nil, func(tServer *testutils.TestServer) {
+		serviceName := tServer.ServiceName()
+
+		server := tServer.NewServer(testutils.NewOpts())
+		defer server.Close()
+
+		subCh := server.GetSubChannel(serviceName)
+		subCh.SetHandler(HandlerFunc(func(ctx context.Context, inbound *InboundCall) {
+			// blackhole all requests
+			inbound.Response().Blackhole()
```

Comment:
If you replace the implementation of `Blackhole` with a `time.Sleep`, the test will still pass.
```
diff --git a/inbound.go b/inbound.go
index a7f7739b..605bc52f 100644
--- a/inbound.go
+++ b/inbound.go
@@ -364,7 +364,8 @@ func (response *InboundCallResponse) SetApplicationError() error {
 // Blackhole indicates that there should be no response and provides
 // an opportunity to clean up resources.
 func (response *InboundCallResponse) Blackhole() {
-       response.cancel()
+       time.Sleep(200 * time.Millisecond)
+       //response.cancel()
 }
```

```
> go test -v -run TestBlackhole
=== RUN   TestBlackhole
--- PASS: TestBlackhole (0.42s)
PASS
ok  	github.com/uber/tchannel-go	0.439s
```

That means the test isn't covering that the blackhole implementation is more efficient than a non-blackhole implementation.

Some suggests on how the test can actually validate it:
 * Before the `Blackhole` call, if you [introspect](https://godoc.org/github.com/uber/tchannel-go#Channel.IntrospectState), you should see that there's a pending message exchange
https://godoc.org/github.com/uber/tchannel-go#ExchangeSetRuntimeState

 * After the `Blackhole` call, there should be no exchanges.

Instead of using a short timeout, you could actually:
 - make a call with a huge timeout (e.g., 1 hour), so we know the exchange isn't being cleared due to timeout
 - after the `Blackhole` command, we can cancel the client context, so the client returns